modules/lang-painless/src/main/java/org/elasticsearch/painless/symbol/WriteScope.java

```diff
@@ -64,21 +66,138 @@ public int getSlot() {
     }
 
     protected final WriteScope parent;
+    protected final ClassWriter classWriter;
+    protected final MethodWriter methodWriter;
+    protected final Label continueLabel;
+    protected final Label breakLabel;
+    protected final Label tryBeginLabel;
+    protected final Label tryEndLabel;
+    protected final Label catchesEndLabel;
     protected final Map<String, Variable> variables = new HashMap<>();
     protected int nextSlot;
 
-    public WriteScope() {
+    protected WriteScope() {
         this.parent = null;
+        this.classWriter = null;
+        this.methodWriter = null;
+        this.continueLabel = null;
+        this.breakLabel = null;
+        this.tryBeginLabel = null;
+        this.tryEndLabel = null;
+        this.catchesEndLabel = null;
         this.nextSlot = 0;
     }
 
-    protected WriteScope(WriteScope parent, int nextSlot) {
+    protected WriteScope(WriteScope parent, ClassWriter classWriter) {
         this.parent = parent;
-        this.nextSlot = nextSlot;
+        this.classWriter = classWriter;
+        this.methodWriter = parent.methodWriter;
+        this.continueLabel = parent.continueLabel;
+        this.breakLabel = parent.breakLabel;
+        this.tryBeginLabel = parent.tryBeginLabel;
+        this.tryEndLabel = parent.tryEndLabel;
+        this.catchesEndLabel = parent.catchesEndLabel;
+        this.nextSlot = parent.nextSlot;
     }
 
-    public WriteScope newScope() {
-        return new WriteScope(this, nextSlot);
+    protected WriteScope(WriteScope parent, MethodWriter methodWriter) {
```

Comment:
jdoc for each of these protected constructors, just a line.

Reply:
Done.